### ISSUE TITLE
feature request: ElectrodeGeometry for channel selection

### ISSUE THREAD

**Issue by bendichter** [flatironinstitute/reactopya_gallery#9]
The `ElectrodeGeometry` widget is looking great. It would be really nice if we could use this widget for selecting channels and have that affect linked widgets. For instance if it was linked to an LFP widget then only the selected channels would be visualized. Features for electrode selection might include:

* double-clicking for single electrode selection
* rectangle drag for multiple selection
* clear all selections
* select all
* `SHIFT` + click or `SHIFT` + drag for adding to selection
* `CTRL` for removing from selection

**Comment by magland** [comment]
Yes, thank you for that list. I will make this high priority. And this will also lead to functionality for communication between widgets / synchronization -- both in javascript and in python.

**Comment by magland** [comment]
@bendichter 
First step of this is complete. I refactored to elec geom widget to make the impl cleaner (separating out a CanvasWidget base class) and then I added prelim selection functionality -- only the basics (click and Ctrl+click). It is live here: http://50.116.50.203:8080/

**Comment by bendichter** [comment]
looks good!

**Comment by magland** [comment]
I've implemented the following, which is similar but quite the same as what you were proposing, @bendichter.

We have a current electrode as well as a set of selected electrodes (2 separate entities)

* click to set current electrode
* ctrl+click or shift+click to select/deselect an electrode (toggle)
* drag rectangle to select multiple
* click anywhere to clear selection

It's live: http://50.116.50.203:8080/
